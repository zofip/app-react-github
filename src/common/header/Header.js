import React, { useEffect, useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  AppBar,
  Toolbar,
  Typography,
  Grid
} from "@material-ui/core";

import MenuAccount from "./menus/MenuAccount";

import { getCookie } from '../../services/serviceCookie';
import { INFO_USER } from '../../util/constants';

const useStyles = makeStyles(theme => ({
  menuButton: {
    marginRight: theme.spacing(2)
  },
  title: {
    flexGrow: 1
  },
  smallDevices: {
    display: "flex !important",
    flexDirection: "row !important"
  }
}));


export default function Header(props) {

  const [infoUser, setInfoUser] = useState(null);

  useEffect(() => {
    setInfoUser(getCookie(INFO_USER));
  }, []);

  const classes = useStyles();
  return (
    <div className={classes.root}>
      <AppBar position="fixed">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Test
          </Typography>
          <div className={classes.smallDevices}>
            {infoUser && (
              <Grid container spacing={2}>
                <Grid item xs>
                  {infoUser.user}
                </Grid>
                <Grid item xs>
                  <MenuAccount />
                </Grid>
              </Grid>
            )}
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
}
