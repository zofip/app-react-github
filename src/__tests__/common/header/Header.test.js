import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../../common/header/Header';

describe('<Header />', () => {
   it('Rendering', () => {
      const header = shallow(<Header />);
      expect(header).toMatchSnapshot();
   });
});