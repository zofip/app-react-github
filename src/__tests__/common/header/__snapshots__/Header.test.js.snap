// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Rendering 1`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-2"
        variant="h6"
      >
        Test
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-smallDevices-3"
      />
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
